From b750a1acd68c7139276a11c205a2d06791719530 Mon Sep 17 00:00:00 2001
From: Marshall Zhan <marshall.zhan.wiwynn@gmail.com>
Date: Thu, 22 Aug 2024 16:43:59 +0800
Subject: [PATCH] Fix the console log file does not rotate

- The file descriptor might be reset after the system power cycle.
- Set the end of the file as the file size.
---
 log-handler.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/log-handler.c b/log-handler.c
index 79b0f99..994d8a8 100644
--- a/log-handler.c
+++ b/log-handler.c
@@ -136,7 +136,7 @@ static int log_create(struct log_handler *lh)
 		warn("Can't open log buffer file %s", lh->log_filename);
 		return -1;
 	}
-	pos = lseek(lh->fd, 0, SEEK_CUR);
+	pos = lseek(lh->fd, 0, SEEK_END);
 	if (pos < 0) {
 		warn("Can't query log position for file %s", lh->log_filename);
 		close(lh->fd);
-- 
2.25.1

